File: movie_recommendation.py
import requests
import openai

# Funktion, um API-Schlüssel aus einer Datei auszulesen
def read_api_key(file_name):
    with open(file_name, 'r') as file:
        return file.read().strip()

# OMDB API-Key und OpenAI API-Schlüssel aus den Dateien lesen
omdb_api_key = read_api_key('omdb_api_key.txt')
openai.api_key = read_api_key('openai_api_key.txt')

# Funktion, um Filmdaten von OMDB abzurufen
def get_movie_data(title, api_key):
    url = f"http://www.omdbapi.com/?t={title}&apikey={api_key}"
    response = requests.get(url)
    return response.json()

# Funktion, um ähnliche Filme von GPT-4 abzurufen
def get_similar_movies(movie_data):
    prompt = f"Ich habe die folgenden Filmdaten: {movie_data}. Welche ähnlichen Filme kannst du mir basierend auf diesen Informationen empfehlen?"

    response = openai.Completion.create(
        model="text-davinci-002",
        prompt=prompt,
        max_tokens=200
    )

    return response.choices[0].text.strip()

# Beispiel-Film abrufen
movie_title = "The Matrix"
movie_data = get_movie_data(movie_title, omdb_api_key)

if movie_data['Response'] == "True":
    similar_movies = get_similar_movies(movie_data)
    print("Ähnliche Filme:", similar_movies)
else:
    print("Film nicht gefunden.")

File: src/components/MovieRecommendation.js
import React, { useState } from "react";
import axios from "axios";

const MovieRecommendation = () => {
  const [movieTitle, setMovieTitle] = useState("");
  const [movieData, setMovieData] = useState(null);
  const [similarMovies, setSimilarMovies] = useState("");

  const omdbApiKey = process.env.REACT_APP_OMDB_API_KEY;
  const openaiApiKey = process.env.REACT_APP_OPENAI_API_KEY;

  const fetchMovieData = async (title) => {
    try {
      const response = await axios.get(
        `http://www.omdbapi.com/?t=${title}&apikey=${omdbApiKey}`,
      );
      setMovieData(response.data);
    } catch (error) {
      console.error("Error fetching movie data:", error);
    }
  };

  const fetchSimilarMovies = async (movieData) => {
    const prompt = `Ich habe die folgenden Filmdaten: ${JSON.stringify(movieData)}. Welche ähnlichen Filme kannst du mir basierend auf diesen Informationen empfehlen?`;

    try {
      const response = await axios.post(
        "https://api.openai.com/v1/chat/completions",
        {
          model: "gpt-4",
          "messages": [
            {"role": "system", "content": "Du bist ein hilfreicher Assistent, der ALLE Filme weltweit kennt und gute Empfehlungen ausspielen kann. Du generierst an Hand von Filmdaten ähnliche Filme. Bitte nenne hier aber nur kurze Stichwort-Listen."},
            {"role": "user", "content": prompt}
          ],
          max_tokens: 200,
        },
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${openaiApiKey}`,
          },
        },
      );

      setSimilarMovies(response.data.choices[0].message.content.trim());
    } catch (error) {
      console.error("Error fetching similar movies:", error);
    }
  };

  const handleSearch = async () => {
    await fetchMovieData(movieTitle);
  };

  const handleRecommendation = async () => {
    if (movieData && movieData.Response === "True") {
      await fetchSimilarMovies(movieData);
    } else {
      setSimilarMovies("Film nicht gefunden.");
    }
  };

  return (
    <div>
      <h1>Film-Empfehlungs-App</h1>
      <input
        type="text"
        value={movieTitle}
        onChange={(e) => setMovieTitle(e.target.value)}
        placeholder="Filmtitel eingeben"
      />
      <button onClick={handleSearch}>Suche Film</button>
      {movieData && (
        <div>
          <h2>
            {movieData.Title} ({movieData.Year})
          </h2>
          <p>{movieData.Plot}</p>
          <button onClick={handleRecommendation}>Ähnliche Filme finden</button>
        </div>
      )}
      {similarMovies && (
        <div>
          <h3>Ähnliche Filme:</h3>
          <p>{similarMovies}</p>
        </div>
      )}
    </div>
  );
};

export default MovieRecommendation;
